vllm/v1/attention/backends/flashinfer.py

```diff
@@ -406,8 +406,9 @@ def _plan(self, attn_metadata: FlashInferMetadata):
                 attn_metadata.decode_wrapper = self._get_decode_wrapper()
                 if not FlashInferBackend.use_trtllm_decode_attention(
                         self._num_decodes, attn_metadata.max_seq_len,
-                        attn_metadata.kv_data_type, attn_metadata.num_qo_heads,
-                        attn_metadata.num_kv_heads, attn_metadata.head_dim):
+                        self.runner.cache_config.cache_dtype,
```

Comment:
@elvischenv Could you explain in the PR description about why we must use `self.runner.cache_config.cache_dtype` instead of `attn_metadata.kv_data_type`?